td-shim-tools/src/tee_info_hash.rs

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+
+use byteorder::LittleEndian;
+use byteorder::ReadBytesExt;
+use core::fmt;
+use scroll::{Pread, Pwrite};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha384};
+use std::fs::File;
+use std::io::Read;
+use std::io::Seek;
+use std::io::SeekFrom;
+use std::mem::size_of;
+use td_layout::metadata::*;
+use zeroize::Zeroize;
+
+pub const SHA384_DIGEST_SIZE: usize = 0x30;
+
+const TDX_METADATA_ATTRIBUTES_EXTEND_MEM_PAGE_ADD: u32 = 0x2;
+
+const TD_INFO_STRUCT_RESERVED_SIZE: usize = 0x70;
+const TDVF_DESCRIPTOR_OFFSET: usize = 0x20;
+const MRTD_EXTENSION_BUFFER_SIZE: usize = 0x80;
+const TDH_MR_EXTEND_GRANULARITY: u64 = 0x100;
+const PAGE_SIZE: u64 = 0x1_000;
+
+pub struct TdvfDesc {
+    pub signature: [u8; size_of::<u32>()],
+    pub length: [u8; size_of::<u32>()],
+    pub version: [u8; size_of::<u32>()],
+    pub numberofsectionentry: [u8; size_of::<u32>()],
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Manifest {
+    #[serde(with = "hex::serde")]
+    pub attributes: [u8; size_of::<u64>()],
+    #[serde(with = "hex::serde")]
+    pub xfam: [u8; size_of::<u64>()],
+    #[serde(with = "hex::serde")]
+    pub mrconfigid: [u8; SHA384_DIGEST_SIZE],
+    #[serde(with = "hex::serde")]
+    pub mrowner: [u8; SHA384_DIGEST_SIZE],
+    #[serde(with = "hex::serde")]
+    pub mrownerconfig: [u8; SHA384_DIGEST_SIZE],
+}
+
+#[repr(C)]
+#[derive(Debug, Pread, Pwrite)]
+pub struct TdInfoStruct {
+    /// TD's attributes
+    pub attributes: [u8; 8],
+    /// TD's XFAM
+    pub xfam: [u8; 8],
+    /// Measurement of the initial contents of the TD
+    pub mrtd: [u8; 48],
+    /// Software-defined ID for non-owner-defined configuration of
+    /// the guest TD
+    pub mrconfig_id: [u8; 48],
+    /// Software-defined ID for the guest TD's owner
+    pub mrowner: [u8; 48],
+    /// Software-defined ID for owner-defined configuration of
+    /// the guest TD
+    pub mrownerconfig: [u8; 48],
+    /// Runtime extendable measurement registers
+    pub rtmr0: [u8; 48],
+    pub rtmr1: [u8; 48],
+    pub rtmr2: [u8; 48],
+    pub rtmr3: [u8; 48],
+    /// Reserved. Must be zero
+    pub reserved: [u8; 112],
```

Comment:
TD_INFO_STRUCT_RESERVED_SIZE ?

Reply:
Just removed the unused traits pread and pwrite of the struct, then the struct can use MACRO.